docassemble/ALToolbox/misc.py

```diff
@@ -179,6 +186,66 @@ def tabbed_templates_html(tab_group_name: str, *pargs) -> str:
     return tabs + tab_content
 
 
+def review_widget(
+    *,
+    up_action:str,
+    down_action:str,
+    review_action:str=None,
+    thumbs_display:str="Did we help you?",
+    review_display:str="Thanks! Let us know what we could do better",
+    submit_review_button:str="Add your review",
+    post_review_display:str="Thank you for your review!",
+) -> str:
+    """
+    A widget that allows people to give a quick review (thumbs up and down, with an optional text
+    component) in the middle of an interview without triggering a page reload.
+
+    If `review_action` is provided, once you press either of the thumbs, a text input screen with
+    a submit button appears, and once the text review is submitted (or after the thumbs, if no
+    `review_action` was provided), a final "thank you" messsage is displayed.
+    
+    @param up_action {string} - the variable name of an event to be executed on the server if the
+        thumbs up is pressed
+    @param down_action {string} - the variable name of an event to be executed on the server i
+        the thumbs down is pressed
+    @param (optional) review_action {string} - the variable name of an event to be execute on the
+        server when someone submits their text review
+    @param (optional) thumbs_display {string} - text displayed to user describing the thumbs
+    @param (optional) review_display {string} - text displayed to user describing the text input
+    @param (optional) submit_review_button {string} - text on the button to submit their text review
+    @param (optional) post_review_display {string} - text displayed to user after review is
+        submitted
+    """
+    js_thumbs_up = f"javascript:altoolbox_thumbs_up_send('{up_action}', {'true' if review_action else 'false'})"
+    js_thumbs_down = f"javascript:altoolbox_thumbs_down_send('{down_action}', {'true' if review_action else 'false'})"
+    widget = f"""
+<div class="card" style="width: 20rem;">
+  <div class="card-body">
+    <p class="al-thumbs-widget">{word(thumbs_display)}</p>
+    <a href="{js_thumbs_up}" id="al-thumbs-widget-up"
+        class="btn btn-md btn-info al-thumbs-widget" aria-label="{word('Thumbs up')}">{fa_icon('thumbs-up', size='md')}</a>
+    <a href="{js_thumbs_down}" id="al-thumbs-widget-down"
+        class="btn btn-md btn-info al-thumbs-widget" aria-label="{word('Thumbs down')}">{fa_icon('thumbs-down', size='md')}</a>
+    """
+    if review_action:
+        review_area_id = review_action + "_area_id"
+        js_review = (
+            f"javascript:altoolbox_review_send('{review_action}', '{review_area_id}')"
+        )
+        widget += f"""
+      <p class="al-review-text al-hidden">{word(review_display)}</p>
```

Comment:
For accessibility, should this (and similar elements) also have `aria-hidden` and then handle that attribute in the js code?

Reply:
`al-hidden` uses `display: none` which is equivalent to `aria-hidden`, and the element doesn't show up in the accessibility tree.